### DISCUSSION TITLE
Error when executing simple script when executing code in IronPython from C#

### DISCUSSION THREAD

**Discussion by Legends** [IronLanguages/ironpython3#1671] [Q&A]
Nuget packages:
- DynamiclanguageRuntime 1.3.3
- IronPython 3.4

The following code executes correctly but also throws a Syntax error

**C#:**
```
ScriptEngine engine = Python.CreateEngine();
ScriptScope scope = engine.CreateScope();
var sb = new StringBuilder("Hello");
scope.SetVariable("sb", sb);

string code = "sb.Append(\"World\")";
ScriptSource source = engine.CreateScriptSourceFromString(code, SourceCodeKind.SingleStatement);

source.Execute(scope);  

Console.WriteLine(sb.ToString()); // HelloWorld
```

**Error:**
![image](https://user-images.githubusercontent.com/966368/225390295-09193b9e-9fde-44a7-96d4-4e99618e721e.png)

Question: Is something wrong with the user code or with the IronPython parser?

**Comment by slozier** [comment]
Hmm, I'm not seeing the syntax error? If you mean `<System.Text.StringBuilder object at 0x000000000000002B [HelloWorld]>` then that's the `repr` of the `StringBuilder` and not a syntax error...

**Reply by Legends** [reply]
Don't know what's a `repr` is?

The SyntaxErrorException is thrown in `Microsoft.Scripting.SourceUnit` (probably unrelated to my issue).
![image](https://user-images.githubusercontent.com/966368/225757983-84162dca-1eff-4a78-9bbd-620850fdcd20.png)

`<System.Text.StringBuilder object at 0x000000000000002B [HelloWorld]>` is printed to the console, when `source.Execute(scope);` has executed.

How can I prevent this?

**Reply by slozier** [reply]
I am still not seeing a `SyntaxError` with the code you provided... Anyway, if you don't want to see the [`repr`](https://docs.python.org/3/library/functions.html#repr) of the `StringBuilder`, either redirect the return value of `sb.Append` to something (e.g. `_ = sb.Append("World")`) or don't use `SourceCodeKind.SingleStatement` (not sure why `SingleStatement` outputs the value but looks like it's been this way for a long time).

**Reply by Legends** [reply]
`_ = sb.Append("World")` The discard did the trick, now I only get `HelloWorld`.

I don't see a SyntaxError too, but when debugging the `SyntaxError` is thrown in  `Microsoft.Scripting.dll`, because `compileCode` is apparently `null`. 

After adding the discard, the code does not jump into the `throw` statement anymore.
